### PR TITLE
Refactor operator to make use of the fdb-kubernetes-monitor API

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Refactoring for making use of the moved constants and structs here: https://github.com/apple/foundationdb/pull/11464

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

## Discussion

I put this as draft until the PR above is merged.

## Testing

Ran unit tests and e2e tests.

## Documentation

-

## Follow-up

-

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -30,6 +30,9 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+// TOOD (johscheuer): Change this back once https://github.com/apple/foundationdb/pull/11464 is merged.
+replace github.com/apple/foundationdb/fdbkubernetesmonitor => github.com/johscheuer/foundationdb/fdbkubernetesmonitor v0.0.0-20240621140641-d991584ea922
```
Comment (johscheuer) [inline go.mod:34]: Remove and update the reference to the main branch.

**Review by johscheuer — COMMENTED** [review]
The fdb-kubernetes-monitor side is merged.

**Review by nicmorales9 — APPROVED** [review]
Looks great, thanks!

File: controllers/update_pod_config.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ package controllers
 
 import (
 	"context"
+	"github.com/apple/foundationdb/fdbkubernetesmonitor/api"
```
[the remark this answers is not in the document]
Comment (johscheuer) [reply controllers/update_pod_config.go:25]: Time to work on https://github.com/FoundationDB/fdb-kubernetes-operator/issues/2049 🙈

**Comment by johscheuer** [comment]
Same failure as in the other ones:

```text
• [FAILED] [413.512 seconds]
Operator when a Pod is unscheduled and another Pod is being replaced [It] should remove the targeted Pod [e2e, pr]
/codebuild/output/src447645687/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator/operator_test.go:583

  [FAILED] Timed out after 300.001s.
  Expected
      <bool>: false
  to be true
  In [It] at: /codebuild/output/src447645687/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/fixtures/fdb_cluster.go:1222 @ 06/24/24 15:56:02.662
------------------------------
```

**Pull request merged**